Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -266,16 +273,36 @@ private string CreateSlug(EventDto eventDto)
         baseSlug = Regex.Replace(baseSlug, @"[^a-z0-9-]", "");
         baseSlug = Regex.Replace(baseSlug, "-+", "-");
 
-        var slug = baseSlug;
+        var existingSlugs = context.Events
+            .Where(e => e.Slug.StartsWith(baseSlug))
+            .Select(e => e.Slug)
+            .ToList();
+
         var counter = 1;
+        if (existingSlugs.Any())
+        {
+            var regex = new Regex($"^{Regex.Escape(baseSlug)}-(\\d+)$");
+            foreach (var slug in existingSlugs)
+            {
+                var match = regex.Match(slug);
+                if (match.Success)
+                {
+                    var currentCounter = int.Parse(match.Groups[1].Value);

```

Comment:
Well, it's not optimal really, I have provided you 1:1 solution like 5 times already that you can filter out first all relevant slugs and then just get count of them, add 1. No need to get all events, iterate through them and count the maximum value.

Reply:
changed